courses/fundamentals_of_ada/adv_060_discriminated_record_types.rst

```diff
@@ -35,32 +35,26 @@ Discriminated Record Types
 Introduction
 ==============
 
-----------------------
-Variant Record Types
-----------------------
-
-* *Variant record type* is a record type where
-
-   + Different objects may have different sets of components (i.e. different variants)
-   + Given object itself may be *unconstrained*
-
-      * Different variants at different times
+----------------------------
+Discriminated Record Types
+----------------------------
 
-* Supported in other languages
+* :dfn:`Discriminated record` type
 
-   + Variant records in Pascal
-   + Unions in C
+   + Different **objects** may have **different** components
+   + All object **still** share the same type
 
-* Variant record offers a kind of storage overlaying
+* Kind of :dfn:`storage overlay`
 
-   + Same storage might be used for one variant at one time, and then for another variant later
-   + Language issue: Ensure this does not provide loophole from type checking
+   + Similar to :C:`union` in C
+   + But preserves **type checking**
+   + And object size **depends** on discriminant
```

Comment:
This is not true for mutable objects

Reply:
this is the introductory slide, we havent introduced mutable types at this point, also we are simply saying that changing the discriminant MAY change the object size, not that it WILL :)